controllers/devworkspace/solver/che_routing.go

```diff
@@ -475,7 +475,7 @@ func provisionMainWorkspaceRoute(cheCluster *chev2.CheCluster, routing *dwo.DevW
 		getServiceURL(wsGatewayPort, dwId, dwNamespace),
 		[]string{"/" + dwId})
 
-	if infrastructure.IsOpenShift() {
+	if infrastructure.IsOpenShift() || utils.IsAccessTokenToPass(cheCluster) {
```

Comment:
We could have default `access_token` for openshift and default `id_token` for kubernetes. This would make these conditions simpler without `IsOpenShift` check. In that case `id_token` would be invalid for openshift and we would need to add some check before.

@tolusha wdyt?

Reply:
@sparkoo 
Probably a stupid question, but what if pass both tokens ?

> We could have default access_token for openshift and default id_token for kubernetes.

It means we will change the default behavior, do you see any pitfalls here?

Reply:
> @sparkoo Probably a stupid question, but what if pass both tokens ?

I just don't like that we would have some token laying around without needing it. Functionally it doesn't matter.

> 
> > We could have default access_token for openshift and default id_token for kubernetes.
> 
> It means we will change the default behavior, do you see any pitfalls here?

It wouldn't change default behavior. It's how it is working now. Access token on openshift (https://github.com/eclipse-che/che-operator/blob/main/pkg/deploy/gateway/oauth_proxy.go#L86) and id token on kubernetes (https://github.com/eclipse-che/che-operator/blob/main/pkg/deploy/gateway/oauth_proxy.go#L117).

Reply:
I see, the let's make conditions simpler without `IsOpenShift` check.

Reply:
> Does it mean that we don't need `IdentityToken` field anymore?

no, because @karatkep wants to use access token on kubernetes, which will not be default.

Reply:
I pushed new changes based on comments above.
@tolusha , @sparkoo , could you please review now and let me know in case any issue/comment?
Thanks in advance!